### PR TITLE
ENH: add extract_unique_points as a GeoSeries property

### PR DESCRIPTION
**Pull request by JamesGardiner**
Implements the shapely extract_unique_points method as part of https://github.com/geopandas/geopandas/issues/2010

### CODE REVIEW

File: geopandas/base.py [hunk 1]
```diff
@@ -618,6 +618,33 @@ def exterior(self):
         # TODO: return empty geometry for non-polygons
         return _delegate_property("exterior", self)
 
+    @property
```
Comment (martinfleis) [inline geopandas/base.py:621]: Any specific reason to make this property (also on an array level)?

Comment (JamesGardiner) [reply geopandas/base.py:621]: I wanted to check this with you actually, and what the preferred approach would be - looking at the current interface, Shapely methods that takes no additional parameters except the input data have been marked as a property e.g. [convex_hull](https://github.com/geopandas/geopandas/blob/979730b6fa1fa9a5f8102b6dcb99ec78a627954b/geopandas/array.py#L516)

Comment (martinfleis) [reply geopandas/base.py:621]: Yeah... I believe that was a mistake made years ago :). Property shall be something that is super cheap, not a method creating new geometry imo. So I'd rather not add more of them, given we already have inconsistent API due to this (e.g. centroid is a property but represetnative_point is a method...). Let's stick to methods for all these new shapely wrappers.

Comment (JamesGardiner) [reply geopandas/base.py:621]: That would be my preference - I'll get this updated. Do we need an issue to track updating the inconsistent API?

Comment (martinfleis) [reply geopandas/base.py:621]: That is already partially happening here #1616. We never properly discussed this and live with the inconsistency but it is on the [roadmap](https://geopandas.org/en/stable/about/roadmap.html#clarity-of-the-api).

File: geopandas/base.py [hunk 2]
```diff
@@ -618,6 +618,32 @@ def exterior(self):
...
+        >>> s
+        0    LINESTRING (0.00000 0.00000, 0.00000 0.00000, ...
+        1    POLYGON ((0.00000 0.00000, 0.00000 0.00000, 1....
+        dtype: geometry
+
+        >>> s.extract_unique_points()
+        0    MULTIPOINT (0.00000 0.00000, 1.00000 1.00000)
+        1    MULTIPOINT (0.00000 0.00000, 1.00000 1.00000)
+        dtype: geometry
+        """
```
Comment (martinfleis) [inline geopandas/base.py:644]: May be useful to crosslink this with `get_coordinates` via See also.

**Review by martinfleis — APPROVED** [review]
Thanks! Just one minor suggestion.

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,17 +4,16 @@
...
-- Added ``offset_curve`` method from shapely to GeoSeries/GeoDataframe. (#2902)
+- Added ``extract_unique_points`` method from shapely to GeoSeries/GeoDataframe (#2915).
+- Added ``hausdorff_distance`` method from shapely to GeoSeries/GeoDataframe (#2909).
+- Added ``delaunay_triangles`` method from shapely to GeoSeries/GeoDataframe (#2907).
+- Added ``concave_hull`` method from shapely to GeoSeries/GeoDataframe (#2903).
+- Added ``offset_curve`` method from shapely to GeoSeries/GeoDataframe (#2902).
 
 New features and improvements:
 
 Bug fixes:
```
Comment (m-richards) [inline CHANGELOG.md:16]:
```suggestion
- ``assert_geodataframe_equal`` now handles GeoDataFrames with no active geometry (#2498)
```
This seems to have been dropped in resolving one of the merge conflicts.

File: geopandas/base.py [hunk 4]
```diff
@@ -733,10 +733,41 @@ def exterior(self):
...
         return _delegate_property("exterior", self)
 
+    def extract_unique_points(self):
+        """Returns a ``GeoSeries`` of MultiPoints representing all
+        distinct vertices of an input geometry.
+
+        Examples
+        --------
+
+        >>> from shapely import LineString, MultiPoint, Point, Polygon
```
Comment (m-richards) [inline geopandas/base.py:743]:
```suggestion
        >>> from shapely import LineString, Polygon
```

**Review by m-richards — APPROVED** [review]

**Pull request merged**